src/app/codemirror/codemirror.component.ts

```diff
@@ -0,0 +1,125 @@
+/**
```

Comment:
Can we refer to npm dep https://www.npmjs.com/package/codemirror instead of copying in source code? If not, I believe this code still needs to live in /third_party as per https://opensource.google/documentation/reference/thirdparty/non-google3#all_non-google_code_goes_into_the_third_party_directory_of_your_project

Reply:
It's probably just a bad name, good catch! It should be probably just  called a "code", it doe refer to CM npm

Reply:
`code-editor.component.ts` / `CodeEditorComponent` is one self-descriptive option for future edits, not blocking this PR.